### DISCUSSION TITLE
Hide User Directory when the user only see his own user

### DISCUSSION THREAD

**Discussion by abdonrd** [directus/directus#11923] [Feature Requests]
When the user has the minimum App Access permissions, he can only see his own user:

<img width="856" alt="Screenshot 2022-03-03 at 01 23 23" src="https://user-images.githubusercontent.com/1007051/156472342-d264f2c1-d9ad-45a5-880a-fc10d916e493.png">

But he can see the entire User Directory view:

<img width="1236" alt="Screenshot 2022-03-03 at 01 28 12" src="https://user-images.githubusercontent.com/1007051/156472754-d5c9b75b-53c3-4c08-b9b5-28c8de5d0c2d.png">

I guess to allow this view where he sees his own profile:

<img width="1238" alt="Screenshot 2022-03-03 at 01 32 21" src="https://user-images.githubusercontent.com/1007051/156473246-4decb748-f1d8-4d67-80b2-4809b0d4ba3c.png">

But I would like to reduce it to just:

<img width="1238" alt="Screenshot 2022-03-03 at 01 35 45" src="https://user-images.githubusercontent.com/1007051/156473482-b1a3dd03-6996-486e-bd9f-e8c528963ccb.png">

Note there is no:
- `User Directory` icon in the main menu.
- Roles list in the second menu.
- No arrow back button to the User Directory.

**Comment by Pachat** [comment]
Also, when the `Users Directory module` is deactivated from the `module-bar`, 
when a user clicks on his avatar at the bottom left, he should not see the roles list in the `module-nav`, not the back button as you said
and => not the link `Users Directory` at the very top of the form.

**Reply by Pachat** [reply]
A solution that already works is to restrict the **`directus_roles`** view access in addition to the `directus_users` view and modify access 
```
{
	"id": {
		"_eq": "$CURRENT_ROLE"
	}
}

```

**Reply by okleinschmidt** [reply]
> A solution that already works is to restrict the **`directus_roles`** view access in addition to the `directus_users` view and modify access
> 
> ```
> {
> 	"id": {
> 		"_eq": "$CURRENT_ROLE"
> 	}
> }
> ```

Can confirm this solution works. I edited the permission of my test role for `Directus Users` and `Directus Roles` as follows:

<img width="634" alt="image" src="https://user-images.githubusercontent.com/6831021/219633962-03f6e7f5-7481-40d6-b98c-d0ab4d913a78.png">

<img width="637" alt="image" src="https://user-images.githubusercontent.com/6831021/219634126-07967b88-ec94-4b17-ac70-d0c4adb8adff.png">

The role is assigned to to users, they can't see each other:

![image](https://user-images.githubusercontent.com/6831021/219634889-a9978eb7-1692-4ecd-a813-e89e7581972b.png)

![image](https://user-images.githubusercontent.com/6831021/219635021-6b0e4d85-c186-4f8d-a6ec-9742c53c9142.png)

Thanks!

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
